workflows/detaxizer.nf

```diff
@@ -105,95 +119,175 @@ workflow DETAXIZER {
 
-    ch_versions = ch_versions.mix(ISOLATE_IDS_FROM_KRAKEN2_TO_BLASTN.out.versions.first())
 
-    //
-    // MODULE: Summarize the kraken2 results and the isolated kraken2 hits
-    //
-    ch_prepare_summary_kraken2 = KRAKEN2_KRAKEN2.out.classified_reads_assignment.join(ISOLATE_IDS_FROM_KRAKEN2_TO_BLASTN.out.classified).map {
-        meta, path1, path2 ->
-            return [ meta, [ path1, path2 ] ]
+        //
+        // MODULE: Isolate the hits for a certain taxa and subclasses
+        //
+        ch_parsed_kraken2_report = PARSE_KRAKEN2REPORT.out.to_filter.map {meta, path -> path}
+
+        KRAKEN2_KRAKEN2.out.classified_reads_assignment.combine(ch_parsed_kraken2_report).set{ ch_combined }
+
+        ISOLATE_KRAKEN2_IDS (
+            ch_combined
+        )
+
+        ch_versions = ch_versions.mix(ISOLATE_KRAKEN2_IDS.out.versions.first())
+
     }
 
-    ch_combined_kraken2 = ch_prepare_summary_kraken2.map {
-        meta, path ->
-            return [ meta +[ id: meta.id.replaceAll("(_R1|_R2)", "") ] , path]
-        }
-        .map {
-            meta, path ->
-                path = path.flatten()
-                return [meta, path]
+    if (params.classification_bbduk) {
+
+
+        //
+        // MODULE: Run bbduk
+        //
+        BBMAP_BBDUK (
+            ch_fastq_for_classification,
+            fasta_bbduk.first()
+        )
+        ch_versions = ch_versions.mix(BBMAP_BBDUK.out.versions.first())
+
+
+        //
+        // MODULE: Run ISOLATE_BBDUK_IDS
+        //
+        ISOLATE_BBDUK_IDS(
+            BBMAP_BBDUK.out.contaminated_reads
+        )
+        ch_versions = ch_versions.mix(ISOLATE_BBDUK_IDS.out.versions.first())
+
+
+    }
+
+    // Prepare MERGE_IDS Channel (with or without merging of IDs)
+
+    if (params.classification_bbduk && params.classification_kraken2){
+
+
+        //
+        // MODULE: Merge IDs
+        //
+        MERGE_IDS(
+            ISOLATE_KRAKEN2_IDS.out.classified_ids.join(
+                ISOLATE_BBDUK_IDS.out.classified_ids, by: [0]
+            ).map{
+                meta, path1, path2 ->
+                    [meta,[path1,path2]]
             }
+        )
+
+
+    } else if (params.classification_bbduk && !params.classification_kraken2){
+
 
-    ch_kraken2_summary = SUMMARY_KRAKEN2(
-        ch_combined_kraken2
+        //
+        // MODULE: Merge IDs
+        //
+        MERGE_IDS(
+            ISOLATE_BBDUK_IDS.out.classified_ids
         )
 
-    ch_versions = ch_versions.mix(ch_kraken2_summary.versions.first())
+
+    } else if (params.classification_kraken2 || (!params.classification_kraken2 && !params.classification_bbduk)){
+
+
+        //
+        // MODULE: Merge IDs
+        //
+        MERGE_IDS(
+            ISOLATE_KRAKEN2_IDS.out.classified_ids
+        )
+
+
+    }
+
+    ch_versions = ch_versions.mix(MERGE_IDS.out.versions.first())
+
+
+    //
+    // MODULE: Summarize the classification results
+    //
+
```

Comment:
You have a lot of superfluous whitespaces here, are they necessary? Above you were generally doing one line spaces, why the doubles here?